test/test_list.rb

```diff
@@ -1,7 +1,10 @@
 # NOTE: following now done in helper.rb (better Readability)
 require 'helper'
 
-ActiveRecord::Base.establish_connection(adapter: "sqlite3", database: ":memory:")
+ActiveRecord::Base.establish_connection(
+  adapter: "sqlite3",
+  database: 'file:memdb1?mode=memory&cache=shared'
```

Comment:
Sorry I got mixed up when running the tests without the SQLite change and seeing no error but that is because it could not indeed test the race condition...
So this change __is__ required to test the race condition but won't break because only 1 thread will be able to access the database